src/RewardsDistributorWorkSubmission.sol

```diff
@@ -0,0 +1,248 @@
+     *  - totalLast24H: the sum of the ring buffer’s most recent 24h.
+     *    We treat that as “locked.”
+     *  - The difference (totalAllSubmissions - totalLast24H) is “unlocked.”
+     *  - We track lastClaimed to ensure we only pay incremental amounts.
+     */
+    function claimRewards(address node) external {
+        require(rewardRatePerUnit != 0, "Rate not set");
+        require(msg.sender == computeRegistry.getNodeProvider(node), "Unauthorized");
+
+        _rollBuckets(node);
+
+        NodeBuckets storage nb = nodeBuckets[node];
+
+        uint256 unlockedNow = nb.totalAllSubmissions - nb.totalLast24H;
+        uint256 claimable = unlockedNow - nb.lastClaimed;
+        if (claimable == 0) {
+            return; // nothing to claim
+        }
+        nb.lastClaimed = unlockedNow;
+
+        uint256 tokensToSend = claimable * rewardRatePerUnit;
+        require(tokensToSend <= rewardToken.balanceOf(address(this)), "Insufficient tokens");
+
+        rewardToken.transfer(node, tokensToSend);
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // "Slash Pending Rewards" Logic
+    // --------------------------------------------------------------------------------------------
+
+    /**
+     * @notice Slashes the pending rewards for a node.
+     *         This is useful for slashing rewards if a node is inactive or misbehaving.
+     *         It resets the node's buckets and totalLast24H to zero.
+     *         Optionally, you can send the slashed tokens to a treasury or burn them.
+     * @param node The address of the node whose pending rewards are to be slashed.
+     * @dev This function can only be called by the REWARDS_MANAGER_ROLE.
+     *      It resets the node's buckets and totalLast24H to zero.
+     */
+    function slashPendingRewards(address node) external onlyRole(REWARDS_MANAGER_ROLE) {
+        _rollBuckets(node);
+        NodeBuckets storage nb = nodeBuckets[node];
+        uint256 pending24h = nb.totalLast24H;
+        if (pending24h == 0) {
+            return; // nothing to slash
+        }
+        for (uint256 i = 0; i < NUM_BUCKETS; i++) {
+            nb.buckets[i] = 0; // reset to zero
+        }
+        nb.totalAllSubmissions -= pending24h; // decrement total
+        nb.totalLast24H = 0; // reset to zero
+        nb.currentBucket = 0; // reset to first bucket
+        nb.lastBucketTimestamp = 0; // reset to zero
+            // Optionally, send the slashed tokens to a treasury or burn them
+            // rewardToken.transfer(treasury, pending24h * rewardRatePerUnit);
+    }
+
+    // --------------------------------------------------------------------------------------------
+    // Optional informational views
+    // --------------------------------------------------------------------------------------------
+
+    function calculateRewards(address node) external view returns (uint256) {
```

Comment:
Is it possible to make this return all rewards by default, and then filter for claimable rewards based on a flag? We need some way to retrieve the user's "total rewards" (including pending + claimable) to display them on the UI

Reply:
I can just make it return a tuple: (total, claimable) - better than doing two transactions.

Reply:
Perfect